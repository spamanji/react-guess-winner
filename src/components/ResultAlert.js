import React from 'react'
import '../styles/newStyles.css';

function ResultAlert(props) {

    const { showAlert, contentMessage, confirmButtonText } = props.alertProps;

    const handleConfirm = (event) => {
        event.preventDefault();
        if (confirmButtonText === 'Play something else') {
            props.showMeMore();
        } else {
            props.restartGame();
        }
    }

    if (showAlert) {
        return (
            <div className='popup'>
                <div className='popup\_inner'>
                    <h3>{contentMessage}</h3>
                    <button onClick={handleConfirm}>{confirmButtonText}</button>
                </div>
            </div>
        )
    }
    return <div></div>;
}

export default ResultAlert;