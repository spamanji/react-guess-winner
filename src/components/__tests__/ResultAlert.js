import React from 'react';
import Enzyme, { shallow, mount, render as enzymeRender } from 'enzyme';
import { fireEvent, render as rtlRender } from '@testing-library/react';
import ResultAlert from '../ResultAlert';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Result Alert Component', () => {
    let mockAlertProps;
    let mockedRestartGame;
    let mockedShowMeMore;

    beforeEach(() => {
        mockAlertProps = {
            showAlert: true,
            contentMessage: 'Test content',
            confirmButtonText: 'Play again'
        }
        mockedShowMeMore = jest.fn();
        mockedRestartGame = jest.fn();
    })

    test('given alert props - returns rendered component - shallow', () => {
        const wrapper = shallow(<ResultAlert alertProps={mockAlertProps} />);

        expect(wrapper.exists()).toBe(true);
        expect(wrapper.find('button').text()).toEqual(mockAlertProps.confirmButtonText);
        expect(wrapper.find('h3').text()).toEqual(mockAlertProps.contentMessage);

    })

    test('given alert props - returns rendered component - enzyme render', () => {
        const wrapper = enzymeRender(<ResultAlert alertProps={mockAlertProps} />);

        expect(wrapper.find('button').text()).toEqual(mockAlertProps.confirmButtonText);
        expect(wrapper.find('h3').text()).toEqual(mockAlertProps.contentMessage);
    })

    test('given alert props - returns rendered component - react testing library render', () => {
        const { getByRole } = rtlRender(<ResultAlert alertProps={mockAlertProps} />);

        expect(getByRole('heading')).toHaveTextContent(mockAlertProps.contentMessage);
        expect(getByRole('button')).toHaveTextContent(mockAlertProps.confirmButtonText);
    })

    test('snapshot test - RTL', () => {
        const { asFragment } = rtlRender(<ResultAlert alertProps={mockAlertProps} />)
        expect(asFragment()).toMatchSnapshot();
    })

    test('given alert props - returns rendered component - mount', () => {
        const wrapper = mount(<ResultAlert alertProps={mockAlertProps} />);

        expect(wrapper.exists()).toBe(true);
        expect(wrapper.find('button').text()).toEqual(mockAlertProps.confirmButtonText);
        expect(wrapper.find('h3').text()).toEqual(mockAlertProps.contentMessage);
    })

    test('click on Play again - a call to \'restartGame\' prop function should be made', () => {
        // arrange
        const wrapper = mount(<ResultAlert alertProps={mockAlertProps} restartGame={mockedRestartGame} />)
        // act
        wrapper.find('button').simulate('click', { preventDefault() { } });
        // assert
        expect(mockedRestartGame).toHaveBeenCalledTimes(1);
    })

    test('click on Play again - a call to \'restartGame\' prop function should be made - RTL ', () => {
        // arrange
        const { getByRole } = rtlRender(<ResultAlert alertProps={mockAlertProps} restartGame={mockedRestartGame} />)
        // act
        fireEvent.click(getByRole('button'));
        // assert
        expect(mockedRestartGame).toHaveBeenCalledTimes(1);
    })


    test('click on Play something else - call to \'showMeMore\' prop function should be made', () => {
        // arrange
        mockAlertProps = {
            showAlert: true,
            contentMessage: 'Test content',
            confirmButtonText: 'Play something else'
        }
        const wrapper = mount(<ResultAlert alertProps={mockAlertProps}
            restartGame={mockedRestartGame} showMeMore={mockedShowMeMore} />)
        // act
        wrapper.find('button').simulate('click', { preventDefault() { } });
        // assert
        expect(mockedShowMeMore).toHaveBeenCalledTimes(1);
    })

    test('click on Play something else - call to \'showMeMore\' prop function should be made - RTL', () => {
        // arrange
        mockAlertProps = {
            showAlert: true,
            contentMessage: 'Test content',
            confirmButtonText: 'Play something else'
        }

        const { getByRole } = rtlRender(<ResultAlert alertProps={mockAlertProps}
            restartGame={mockedRestartGame} showMeMore={mockedShowMeMore} />)
        // act
        fireEvent.click(getByRole('button'));
        // assert
        expect(mockedShowMeMore).toHaveBeenCalledTimes(1);
        expect(mockedRestartGame).toHaveBeenCalledTimes(0);
    })


})