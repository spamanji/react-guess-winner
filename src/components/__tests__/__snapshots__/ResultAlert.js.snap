// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Alert Component snapshot test - RTL 1`] = `
<DocumentFragment>
  <div
    class="popup"
  >
    <div
      class="popup\\\\_inner"
    >
      <h3>
        Test content
      </h3>
      <button>
        Play again
      </button>
    </div>
  </div>
</DocumentFragment>
`;
